src/ls_server.rs

```diff
@@ -251,34 +252,36 @@ impl LsService {
     }
 
     fn init(&self, id: usize, init: InitializeParams) {
-        let result = InitializeCapabilities {
+        let result = InitializeResult {
             capabilities: ServerCapabilities {
-                textDocumentSync: DocumentSyncKind::Incremental as usize,
-                hoverProvider: true,
-                completionProvider: CompletionOptions {
-                    resolveProvider: true,
-                    triggerCharacters: vec![".".to_string()],
-                },
+                text_document_sync: Some(TextDocumentSyncKind::Incremental),
```

Comment:
Ugh, I also am quite averse to that suggestion of adding None/Missing variants. Feels very wrong: it makes it impossible to specify a non-null `Foo`, which I think is an inferior abstraction. I don't think expressive power should be sacrificed just so that we save some minor boilerplate code when matching against `Option<Foo>` .